packages/process/src/node/raw-process.ts

```diff
@@ -50,31 +51,31 @@ export interface RawProcessFactory {
     (options: RawProcessOptions | RawForkOptions): RawProcess;
 }
 
-/**
- * A Node stream like `/dev/null`.
- *
- * Writing goes to a black hole, reading returns `EOF`.
- */
-class DevNullStream extends stream.Duplex {
-    // tslint:disable-next-line:no-any
-    _write(chunk: any, encoding: string, callback: (err?: Error) => void): void {
-        callback();
-    }
-
-    _read(size: number): void {
-        // tslint:disable-next-line:no-null-keyword
-        this.push(null);
-    }
-}
-
 @injectable()
 export class RawProcess extends Process {
 
+    /**
+     * @deprecated use `inputStream` instead.
+     */
     readonly input: stream.Writable;
+
+    /**
+     * @deprecated use `outputStream` instead.
+     */
     readonly output: stream.Readable;
+
+    /**
+     * @deprecated use `errorStream` instead.
+     */
     readonly errorOutput: stream.Readable;
+
     readonly process: ChildProcess;
 
+    // shim to not break APIs
+    get outputStream() { return this.output; }
```

Comment:
Hmm I think it should be the other way around:  outputStream errorStream and inputStream are the "real" fields, and output, input, errorOutput are the getter shims (since the latter are the deprecated ones).

Reply:
Also, maybe it can be done as a separate cleanup, pushed on its own?

Reply:
> Also, maybe it can be done as a separate cleanup, pushed on its own?

Are you talking about the addition of streams to the abstract `Process` class?

Reply:
> Are you talking about the addition of streams to the abstract `Process` class?

Maybe I don't quite understand, but it just looks like a rename from output to outputStream, stdin to stdinStream etc?  If I am mistaken, could you precise why this change is needed?

Reply:
To normalize the process abstract class, before we had to check which implementation (Process/Terminal) we were using before trying to write/read.

Now since the base class `Process` requires the fields, subclasses just implement it.

I think at some point it helped me with my changes, but with the rewrites maybe it is not directly helpful anymore. I still think it is a good thing.